pkg/util/clientauthorizer/arm.go

```diff
@@ -98,7 +99,12 @@ func (a *arm) refresh() {
 func (a *arm) refreshOnce() error {
 	now := a.now()
 
-	req, err := http.NewRequest(http.MethodGet, "https://management.azure.com:24582/metadata/authentication?api-version=2015-01-01", nil)
+	environment, err := azureenvironment.AzureEnvFromEnvVar()
```

Comment:
store instancemetadata.InstanceMetadata as a field inside the arm type and read the environment from there

Reply:
Done.

To make sure the `arm_test.go` tests still worked, I added a function, `instancemetadata.NewFake`, which doesn't require environment variables to be set. I didn't put it in its own subpackage because `instanceMetadata` is unexported. Open to suggestions about this.